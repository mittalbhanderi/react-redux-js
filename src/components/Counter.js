import React, { PureComponent } from 'react'
import PropTypes from 'prop-types';

class Counter extends PureComponent {

constructor(props) {
  super(props);
  this.incrementIfOdd = this.incrementIfOdd.bind(this);
}

  static propTypes = {
    value: PropTypes.number.isRequired,
    onIncrement: PropTypes.func.isRequired,
    onDecrement: PropTypes.func.isRequired,
    onAsyncIncrement: PropTypes.func.isRequired
  }

  incrementIfOdd(){
    if (this.props.value % 2 !== 0) {
      this.props.onIncrement()
    }
  }

  render() {
    const { value, onIncrement, onAsyncIncrement, onDecrement } = this.props
    return (
      <div>
        Clicked: {value} times
              {' '}
        <button onClick={onIncrement}>
          +
              </button>
        {' '}
        <button onClick={onDecrement}>
          -
              </button>
        {' '}
        <button onClick={this.incrementIfOdd}>
          Increment if odd
              </button>
        {' '}
        <button onClick={onAsyncIncrement}>
          Increment async
              </button>
      </div>
    )
  }
}

export default Counter