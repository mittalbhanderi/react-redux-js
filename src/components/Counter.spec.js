import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Counter from './Counter';

function setup(counter) {

  const component = shallow(
    <Counter value={counter} onIncrement={jest.fn()}
      onDecrement={jest.fn()}
      onAsyncIncrement={jest.fn()} />
  )

  return {
    component: component,
    addButton: component.findWhere(n => n.type() === 'button' && n.contains('+')),
    negateButton: component.findWhere(n => n.type() === 'button' && n.contains('-'))
  }
}

test('Counter component snapshot', () => {
  const component = renderer.create(
    <Counter
      value={0}
      onIncrement={jest.fn()}
      onDecrement={jest.fn()}
      onAsyncIncrement={jest.fn()}></Counter>,
  );

  let tree = component.toJSON();
  tree.props.incrementIfOdd = jest.fn();
  expect(tree).toMatchSnapshot();

});


describe('Counter Component', () => {
  it('should display counter component', () => {
    const { component } = setup(1);

    expect(component.text()).toMatch("Clicked: 1 times + - Increment if odd Increment async");
  });

});




