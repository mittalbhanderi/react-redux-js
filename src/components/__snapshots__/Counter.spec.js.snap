// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component snapshot 1`] = `
<div
  incrementIfOdd={[MockFunction]}
>
  Clicked: 
  0
   times
   
  <button
    onClick={[MockFunction]}
  >
    +
  </button>
   
  <button
    onClick={[MockFunction]}
  >
    -
  </button>
   
  <button
    onClick={[Function]}
  >
    Increment if odd
  </button>
   
  <button
    onClick={[MockFunction]}
  >
    Increment async
  </button>
</div>
`;
